psi4/driver/p4util/python_helpers.py

```diff
@@ -75,7 +75,16 @@ def pybuild_basis(mol, key=None, target=None, fitrole='ORBITAL', other=None, pur
         core.print_out(basisdict['message'])
 
     psibasis = core.BasisSet.construct_from_pydict(mol, basisdict, puream)
-    return psibasis
+    ecpbasis = None
+    if 'ecp_shell_map' in basisdict:
+        ecpbasis = core.BasisSet.construct_ecp_from_pydict(mol, basisdict, puream)
+
+    if key == 'BASIS':
+        # For orbitals basis sets, we need to return ECP also
+        return psibasis, ecpbasis
```

Comment:
For example, I think this will break most of the Psi4NumPy like scripts as the basis is usually built explicitly.

Since we have a bit of time before this goes in (as we are waiting for 1.1), I wouldn't mind cleaning up this particular aspect. Reading through your code (and as you mention) it looks like the ECP info stuff inside a conventional basis object should be fairly straightforward. This may help with the book keeping as well. Of course @loriab would be great to comment on this.